### ISSUE TITLE
False Positive For `camelcase` when destructuring into variables starting with _

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#9710]
**Tell us about your environment**

* **ESLint Version:** 4.13.0
* **Node Version:** 8.9.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 8,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "camelcase": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const test = {
    _a: 1
};
const _b = asdf;
const { _a } = blah;
```

ESLint.org demo

**What did you expect to happen?**
No reported errors.

**What actually happened? Please include the actual, raw output from ESLint.**

`5:9 - Identifier '_a' is not in camel case. (camelcase)`

**Comment by epmatsw** [comment]
Drat, dupe of https://github.com/eslint/eslint/issues/9700
